sklearn/datasets/_california_housing.py

```diff
@@ -48,8 +49,17 @@
 logger = logging.getLogger(__name__)
 
 
+def _convert_data_dataframe(caller_name, data, target, feature_names, target_names):
```

Comment:
this probably should live in ``sklearn/datasets/base.py``

Reply:
Just moved it after chatting with @wconnell :)
(I left it here temporarily since I wasn't sure it would work for all cases; it's in `_base.py` in the most recent commit)